src/unauthenticated_stream/chacha20.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use {chacha, error, polyfill, unauthenticated_stream};
+use polyfill::slice::u32_from_le_u8;
+
+/// ChaCha20 unauthenticated stream cipher.
+///
+/// The keys are 256 bits long and the nonces are 128 bits long, rather then
+/// the usual 96 bits. The first 32 bits of the 128-bit nonce represents an
+/// explicit initial counter parameter.
+pub static CHACHA20: unauthenticated_stream::Algorithm =
+                                        unauthenticated_stream::Algorithm {
+    key_len: chacha::KEY_LEN_IN_BYTES,
+    init: chacha20_init,
+    xor_keystream: chacha20_xor_keystream,
+    id: unauthenticated_stream::AlgorithmID::CHACHA20,
+    max_input_len: max_input_len!(CHACHA20_BLOCK_LEN, CHACHA20_OVERHEAD_BLOCKS_PER_NONCE),
+};
+
+const CHACHA20_BLOCK_LEN: u64 = 64;
+const CHACHA20_OVERHEAD_BLOCKS_PER_NONCE: u64 = 1;
```

Comment:
These need to be defined in a single place, not duplicated between the ChaCha20-Poly1305 AEAD code and the unauthenticated ChaCha20 code.

Reply:
ACK.

Reply:
Done.